src/Umbraco.Core/Services/Implement/FileService.cs

```diff
@@ -539,6 +544,12 @@ public IEnumerable<ITemplate> GetTemplateChildren(int masterTemplateId)
         /// <param name="userId"></param>
         public void SaveTemplate(ITemplate template, int userId = Constants.Security.SuperUserId)
         {
+            if (template.Name != null && template.Name.Length > 255)
+            {
+                throw new InvalidOperationException("Name cannot be more than 255 characters in length.");
```

Comment:
This one is a little trickier, as `template.Name` isn't an argument and therefore shouldn't throw the same exceptions as `CreateTemplateWithIdentity`.

It should either throw an `ArgumentException` referencing `template` or keep this `InvalidOperationException`. Either way: the message should indicate it can't be null, empty/white space or more than 255 characters in length.